### PR TITLE
Bug 2086519: AUTH-133: test/extended/security/scc: comply to pod security admission 

### PR DESCRIPTION
**Pull request by stlaz**
Obsoletes #27154 

/assign @ibihim

### CODE REVIEW

File: test/extended/builds/build_pruning.go [hunk 1]
```diff
@@ -27,7 +29,7 @@ var _ = g.Describe("[sig-builds][Feature:Builds] prune builds based on settings
 		failedBuildConfig     = filepath.Join(buildPruningBaseDir, "failed-build-config.yaml")
 		erroredBuildConfig    = filepath.Join(buildPruningBaseDir, "errored-build-config.yaml")
 		groupBuildConfig      = filepath.Join(buildPruningBaseDir, "default-group-build-config.yaml")
-		oc                    = exutil.NewCLI("build-pruning")
+		oc                    = exutil.NewCLIWithPodSecurityLevel("build-pruning", admissionapi.LevelBaseline)
```
Comment (ibihim) [inline test/extended/builds/build_pruning.go:32]: In general, I am wondering if we need to expose the complexity of PodSecurityLevel to everyone. Maybe we could add `admissionapi.LevelBaseline` to every `exutil.NewCLI` by default and only set `exutil.NewCLIWithPodSecurityLevel` in the case of `admissionapi.LevelRestricted` or `admissionapi.LevelPrivileged`.

Comment (s-urbaniak) [reply test/extended/builds/build_pruning.go:32]: The default must be the most secure one (restricted). only once your e2e workload needs broader permissions you have to make a conscious choice (baseline or restricted). This behaviour also aligns with upstream.

**Pull request merged**